chain/network-primitives/src/network_protocol/edge.rs

```diff
@@ -141,6 +141,12 @@ impl Edge {
         Edge(Arc::new(edge))
     }
 
+    pub fn with_removal_info(self, ri: Option<(bool, Signature)>) -> Edge {
+        let mut edge = self.0.as_ref().clone();
+        edge.removal_info = ri;
+        Edge(Arc::new(edge))
+    }
```

Comment:
```suggestion
    pub fn with_removal_info(mut self, ri: Option<(bool, Signature)>) -> Edge {
        let inner = Arc::make_mut(&mut self.0);
        inner.removal_info = ri;
        self
    }
```

will avoid clone in the typical usage.

Let's maybe move this next to `new`, as this is sufficiently constructor-shaped?

Reply:
done